rest/route/project.go

```diff
@@ -656,6 +656,82 @@ func (h *projectRepotrackerHandler) Run(ctx context.Context) gimlet.Responder {
 	return gimlet.NewJSONResponse(struct{}{})
 }
 
+////////////////////////////////////////////////////////////////////////
+//
+// DELETE /rest/v2/projects/{project_id}
+
+type projectDeleteHandler struct {
+	projectName string
+	sc          data.Connector
+}
+
+func makeDeleteProject(sc data.Connector) gimlet.RouteHandler {
+	return &projectDeleteHandler{
+		sc: sc,
+	}
+}
+
+func (h *projectDeleteHandler) Factory() gimlet.RouteHandler {
+	return &projectDeleteHandler{
+		sc: h.sc,
+	}
+}
+
+func (h *projectDeleteHandler) Parse(ctx context.Context, r *http.Request) error {
+	h.projectName = gimlet.GetVars(r)["project_id"]
+	return nil
+}
+
+func (h *projectDeleteHandler) Run(ctx context.Context) gimlet.Responder {
+	project, err := dbModel.FindMergedProjectRef(h.projectName)
```

Comment:
In the case of routes, we don't need to look at the merged project ref (which is the project ref and the repo ref settings combined), because Hidden and UseRepoSettings can only be defined at the branch level, so FindOneProjectRef would be sufficient (you'll also want to check for that that project isn't nil)